packages/flutter/lib/src/widgets/page_view.dart

```diff
@@ -370,7 +384,8 @@ class _PagePosition extends ScrollPositionWithSingleContext implements PageMetri
   double get _initialPageOffset => math.max(0, viewportDimension * (viewportFraction - 1) / 2);
 
   double getPageFromPixels(double pixels, double viewportDimension) {
-    final double actual = math.max(0.0, pixels - _initialPageOffset) / math.max(1.0, viewportDimension * viewportFraction);
+    assert(viewportDimension > 0.0);
+    final double actual = math.max(0.0, pixels - _initialPageOffset) / (viewportDimension * viewportFraction);
```

Comment:
`math.max(1.0, viewportDimension * viewportFraction)` -> `(viewportDimension * viewportFraction)`

This fixes the issue when the viewport is less than 1.0 although it is not common.